common/io/Serial.cpp

```diff
@@ -27,6 +27,7 @@
 #include <signal.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <sys/file.h>
```

Comment:
Has sys/file.h existed forever, so we're not breaking UUCP builds which don't have it?

Reply:
I'm not sure, I'll investigate.  However the header is checked by the configure script, and used without any #ifdefs in one other place (KarateLight).

Reply:
Ah okay, obviously that plugin breaking wouldn't have quite so much impact, but the plugin has been around for a while and no-one has complained so we're probably good...

Reply:
he flock call itself conforms to 4.4BSD, i.e. it's been around since 1997.  So I think we're pretty safe here.

I could add guards around the #include (ifdef HAVE_SYS_FILE_H), but if it's worked so far then I don't see any reason to add the extra complexity.   On inspection, I'm not actually sure why KarateLight includes this header at all.

Reply:
Yeah very good point about the fact it's worked!

I'm not sure about KarateLight either; @cpresser ?

Reply:
Unfortunately, I am not sure what to answer here. My C++ is quite rusty, I have not used it in years and sure have no idea which include contains which functionality.

I recall that I did include file locking of the serial-port-device in the karatelight code. A quick search turned up this piece of code here:
https://github.com/OpenLightingProject/ola/blob/master/plugins/karate/KarateLight.cpp#L121

Does that help?

Reply:
The `#include <sys/file.h>` has been there since the first version of KarateDevice.cpp (f69a26b065adc853c3f7273d298c1589b1a5df68), but the `flock` call is in a different file (KarateLight.cpp).  I suspect the `flock` calls were originally in KarateDevice.cpp and got moved.

It does mean, however, that there's not only the #include but also a real use of flock() in OLA already without any tests!  Therefore we're probably pretty safe using it for all serial devices.  Nevertheless, we'll keep the tests and the conditionals :)